api/nodes/serializers.py

```diff
@@ -124,8 +131,20 @@ def update(self, instance, validated_data):
         the request to be in the serializer context.
         """
         assert isinstance(instance, Node), 'instance must be a Node'
+        auth = self.get_user_auth(self.context['request'])
         for attr, value in validated_data.items():
-            setattr(instance, attr, value)
+            if attr == 'tags':
+                old_tags = set([tag._id for tag in instance.tags])
+                if value:
+                    current_tags = set(value)
+                else:
+                    current_tags = set()
+                for new_tag in (current_tags - old_tags):
+                    instance.add_tag(new_tag, auth=auth)
+                for deleted_tag in (old_tags - current_tags):
```

Comment:
No, that should work. And there *should* be a test for it… https://github.com/CenterForOpenScience/osf.io/pull/4093/files#diff-14619b95d1761b6ed098d90acc33b79eR1132

It's not a collection, so this is the only real way to update a list field. We *could* make it a collection, which was the other option, but then we have to have include working in order to get the tags inside the response, and it's fairly heavyweight just for tags.